templates/serverpod_templates/projectname_server_upgrade/Dockerfile

```diff
@@ -1,29 +1,23 @@
-# If you update the dart version, make sure the image is
-# compatible with the busybox image.
 FROM dart:3.2.5 AS build
 
 WORKDIR /app
 COPY . .
 
 RUN dart pub get
-RUN dart compile exe bin/main.dart -o bin/main
+RUN dart compile exe bin/main.dart -o bin/server
 
-# If you update the busybox version, make sure the image is
-# compatible with the dart image.
-FROM busybox:1.36.1-glibc
-
-ENV runmode=development
-ENV serverid=default
-ENV logging=normal
-ENV role=monolith
+FROM scratch
 
 COPY --from=build /runtime/ /
-COPY --from=build /app/bin/main /app/bin/main
+COPY --from=build /app/bin/server server
 COPY --from=build /app/config/ config/
 COPY --from=build /app/web/ web/
+COPY --from=build /app/migrations/ migrations/
+
 
 EXPOSE 8080
 EXPOSE 8081
 EXPOSE 8082
```

Comment:
Not necessarily part of this PR, but shouldn't the ports come from the yaml config files, with these here just being the defaults? We are trying to go down this route in our WIP PR over at #2124 - right now we are missing a source of truth, as the config yaml is only used for the dart code and we need to manually keep track of what needs to be configured where.

Reply:
Good point! But indeed another PR for dealing with that.